api/v1alpha2/istio_merge.go

```diff
@@ -23,6 +23,17 @@ const (
 	requestsField         = "requests"
 )
 
+// compatibility environment variables of Istio
+// they are supposed to be feature flags, to keep the compatibility with previous Istio versions in case of breaking changes
+// should be updated with every Istio bump according to the release notes
+// requires update of integration tests
+// current env comes from: Istio 1.21
+var compatibilityEnvs = map[string]string{
+	"ENABLE_EXTERNAL_NAME_ALIAS": "false",
```

Comment:
As far as I know we already have this feature flag with `false` only for one cluster right now by applying the annotation. Why do we have it here as default for compatibility mode? I think we ned to exclude it from compatibility mode.

Reply:
But then, if a new cluster want to go into the compatibility mode, it has to also apply the annotation to fully achieve that, which is counter intuitive. Current behaviour let you use annotation to apply only this one env variable with false value, but if you decide to to turn compatibility mode on, then it's applied as well.